lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -1640,8 +1653,18 @@ void StmtEmitter::emitExpression(Value exp,
                     outBuffer.end());
     outBuffer.resize(blockDeclarationInsertPointIndex);
     for (auto *expr : tooLargeSubExpressions) {
+      SmallPtrSet<Operation *, 8> emittedExprs;
       emitDeclarationForTemporary(expr);
-      os << ";\n";
+
+      // If this is a constant, we have to immediately assign its value since
+      // as required per the `localparam` construct they are emitted as.
+      if (isConstantExpression(expr)) {
```

Comment:
Should this special case go in `emitDeclarationForTemporary`?

Reply:
Yes this simplifies things. Also removes the need for the special case you commented on above.